platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/report/generic/PerRootTabRenderer.java

```diff
@@ -256,4 +262,123 @@ protected void render(TestTreeModel.PerRootInfo info, SimpleHtmlWriter htmlWrite
         }
     }
 
+    @SuppressWarnings({"MethodMayBeStatic", "UnusedReturnValue"})
+    public static final class ForMetadata extends PerRootTabRenderer {
+        private final MetadataRendererRegistry metadataRendererRegistry;
+
+        public ForMetadata(String rootName, MetadataRendererRegistry metadataRendererRegistry) {
+            super(rootName);
+            this.metadataRendererRegistry = metadataRendererRegistry;
+        }
+
+        @Override
+        protected void render(TestTreeModel.PerRootInfo info, SimpleHtmlWriter htmlWriter) throws IOException {
+            htmlWriter.startElement("div").attribute("id", "metadata");
```

Comment:
:x: We can't use `id`s in this code as it can be rendered multiple times in aggregate reports. This needs to use a `class`.